openedx/core/djangoapps/agreements/serializers.py

```diff
@@ -17,4 +17,17 @@ class IntegritySignatureSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = IntegritySignature()
-        fields = ('username', 'course_id', 'created_at')
+
+
+class LTIPIISignatureSerializer(serializers.ModelSerializer):
+    """
+    Serializer for LTIPIISignature model
+    """
+    username = serializers.CharField(source='user.username')
+    course_id = CourseKeyField(source='course_key')
+    lti_tools = serializers.JSONField()
+    created_at = serializers.DateTimeField(source='created')
+
+    class Meta:
+        model = LTIPIISignature()
```

Comment:
I'm not really sure why `IntegritySignatureSerializer` works, but the `model` attribute should not instantiate the model class. It should just be a reference to the class.

```suggestion
        model = LTIPIISignature
```
Please see the Django Rest Framework documentation [here](https://www.django-rest-framework.org/api-guide/serializers/#modelserializer).